go/consensus/tendermint/apps/roothash/transactions.go

```diff
@@ -379,17 +381,24 @@ func (app *rootHashApplication) submitMsg(
 
 	// Queue message.
 	inMsg := &message.IncomingMessage{
-		ID:     meta.NextSequenceNumber,
-		Caller: ctx.CallerAddress(),
-		Tag:    msg.Tag,
-		Fee:    msg.Fee,
-		Tokens: msg.Tokens,
-		Data:   msg.Data,
+		ID:          meta.NextSequenceNumber,
+		Caller:      ctx.CallerAddress(),
+		Tag:         msg.Tag,
+		Fee:         msg.Fee,
+		Tokens:      msg.Tokens,
+		Transaction: msg.Transaction,
+		Data:        msg.Data,
 	}
 	if err = state.SetIncomingMessageInQueue(ctx, rtState.Runtime.ID, inMsg); err != nil {
 		return err
 	}
 
+	if msg.Transaction != nil {
```

Comment:
Huh? Why are you changing the consensus layer? You probably want to put this somewhere in `go/worker/compute` where it gets incoming messages, e.g.:
https://github.com/oasisprotocol/oasis-core/blob/9f399d01a3c17358a0bf278a3d6a5392deda20e5/go/worker/compute/executor/committee/node.go#L934-L938

Reply:
ah yeah that's a good idea to do it for the entire batch. has to be a little before this exact point you link to though, because this point comes after the node fully prepares a [][]byte of transaction input data. we'll need to replicate the transactions in the txpool before that